pubtools/pulplib/_impl/client/repo_lock.py

```diff
@@ -0,0 +1,161 @@
+import json
+import random
+import string
+from time import sleep
+
+from datetime import datetime, timedelta
+from more_executors.futures import f_map
+
+
+LOCK_CLAIM_STR = "lock-claim-"
+
+LOCK_VALID_FROM_OFFSET = 10
+LOCK_EXPIRATION_OFFSET = 3600
```

Comment:
I'm pretty sure that the expiration should be part of the public API.  This could be a default, but only the caller has an idea of how long they'd expect to be holding the lock, so it'd be best if they can set a smaller expiration in some cases.

For example in the maintenance-on/off case, it should only need the lock for a few seconds.  The default would be safe but it would also be overkill.

The other magic numbers here probably don't need to be public API though we generally make them tunable by "hidden" environment variables, so that we can at least tweak them in emergency scenarios without code changes if that turns out to be needed.